src/main/java/org/vaadin/addon/leaflet/util/PointField.java

```diff
@@ -49,15 +48,15 @@ protected void prepareDrawing() {
             map.removeComponent(marker);
             marker = null;
         }
-        map.addClickListener(drawListener);
+        markerRegisration = map.addClickListener(drawListener);
     }
 
     @Override
     protected void prepareViewing() {
-    	map.removeClickListener(drawListener);
-    	map.removeClickListener(editClickListener);
-    	if(marker != null) {
-    		marker.removeDragEndListener(editDragEndListener);
+        drawRegistration.remove();
```

Comment:
Don't have the project open ATM, should there be a null check? Can the field go for "view mode" directly if a populated readonly property is set (I don't actually know if this is possible currently with V8)?